### PR TITLE
Split modules

### PR DESCRIPTION
**Pull request by annie-anna**
- `Action.Outcome` -> `Outcome.re`

- `JSUtil.ModKey`, `JSUtil.ModKeys` -> `ModKeys.ModKey`, `ModKeys.re`

- `JSUtil.Key`, `JSUtil.KeyCombo` -> `KeyCombo.Key`, `KeyCombo.re`

- `JSUtil.MoveKey` -> `MoveKey.re`

- `Update.Action` -> `ModelAction.re`

### CODE REVIEW

**Comment by annie-anna** [comment]
Do I need to split `JSUtil.MoveKey`  and `Update.Action` ?

**Comment by cyrus-** [comment]
yes. let's rename "Update.Action" to "ModelAction" to make it clear that these are actions that operate on models, whereas EditActions are those that operate on edit states

File: src/hazelweb/util/KeyCombo.re [hunk 1]
```diff
@@ -0,0 +1,275 @@
+module Js = Js_of_ocaml.Js;
+module Dom_html = Js_of_ocaml.Dom_html;
+
+let get_code = (evt: Js.t(Dom_html.keyboardEvent)) =>
+  Js.to_string(Js.Optdef.get(evt##.code, () => assert(false)));
+
+let get_key = (evt: Js.t(Dom_html.keyboardEvent)) =>
+  Js.to_string(Js.Optdef.get(evt##.key, () => assert(false)));
+
+module Key = {
```
Comment (cyrus-) [inline src/hazelweb/util/KeyCombo.re]: let's move this to its own file

File: src/hazelweb/util/KeyCombo.re [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+      String.equal(String.uppercase_ascii(key), String.uppercase_ascii(k));
+    };
+
+  let matches = (k, evt: Js.t(Dom_html.keyboardEvent)) => {
+    let recognition_methods = k.recognition_methods;
+    ListUtil.any(recognition_methods, recognize(evt));
+  };
+};
+
+module Details = {
```
Comment (cyrus-) [inline src/hazelweb/util/KeyCombo.re]: Details.t should really be KeyCombo.t I think? (@dmoon1221 -- why is this called `Details`?)

File: src/hazelweb/util/KeyCombo.re [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  let ctrl_alt_i = ctrl_alt(Key.the_key("i"));
+  let ctrl_alt_k = ctrl_alt(Key.the_key("k"));
+  let ctrl_alt_j = ctrl_alt(Key.the_key("j"));
+  let ctrl_alt_l = ctrl_alt(Key.the_key("l"));
+  let meta_z = ctrl(Key.the_key("z"));
+  let meta_shift_z = ctrl_shift(Key.the_key("Z"));
+};
+
+[@deriving sexp]
+type t =
```
Comment (cyrus-) [inline src/hazelweb/util/KeyCombo.re]: this type should be called HazelKeyCombos.t in a separate file I think -- it enumerates the key combos that Hazel listens for

File: src/hazelweb/util/ModKeys.re [hunk 4]
```diff
@@ -0,0 +1,67 @@
+module Js = Js_of_ocaml.Js;
+module Dom_html = Js_of_ocaml.Dom_html;
+
+module ModKey = {
```
Comment (cyrus-) [inline src/hazelweb/util/ModKeys.re]: move this to a separate file

**Review by cyrus- — CHANGES_REQUESTED** [review]

**Comment by cyrus-** [comment]
also let's make a folder `keyboard` for all the key handling related modules

File: src/hazelweb/util/KeyCombo.re (hunk 2 above)
Comment (annie-anna) [reply src/hazelweb/util/KeyCombo.re]: so I should remove the module ``Details``?

Comment (cyrus-) [reply src/hazelweb/util/KeyCombo.re]: the contents of the Details module should become the KeyCombo.t module. current KeyCombo.t type should become HazelKeyCombo.t

**Comment by cyrus-** [comment]
brown is a little dark and hard to distinguish. let's use a teal color like #216565, which seems to be fine against both the exp and pat backgrounds.

File: src/hazelcore/Outcome.re [hunk 5]
```diff
@@ -0,0 +1,9 @@
+type t('success) =
```
Comment (cyrus-) [inline src/hazelcore/Outcome.re]: Let's call this module ActionOutcome, rather than just Outcome, to make it more clear that it is related to actions

**Review by cyrus- — CHANGES_REQUESTED** [review]

**Review by cyrus- — APPROVED** [review]
looks good, thanks!

**Pull request merged**